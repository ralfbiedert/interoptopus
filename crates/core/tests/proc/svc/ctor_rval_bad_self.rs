use interoptopus::{ffi, ffi_service, ffi_type};

#[ffi_type]
enum Error {
    Something,
}

#[ffi_type(service)]
struct Service;

#[ffi_service]
impl Service {
    pub fn new() -> ffi::Result<u8, Error> {
        ffi::Ok(0)
    }
}

#[allow(dead_code)]
fn main() {}
