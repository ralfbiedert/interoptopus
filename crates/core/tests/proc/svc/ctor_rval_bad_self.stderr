error[E0080]: evaluation panicked: This method looks like a constructor, but does not return ffi::Result<Self, _>
  --> tests/proc/svc/ctor_rval_bad_self.rs:13:12
   |
13 |     pub fn new() -> ffi::Result<u8, Error> {
   |            ^^^ evaluation of `_` failed here

error[E0308]: mismatched types
  --> tests/proc/svc/ctor_rval_bad_self.rs:13:12
   |
13 |     pub fn new() -> ffi::Result<u8, Error> {
   |            ^^^
   |            |
   |            expected `*const Service`, found `*mut u8`
   |            expected due to the type of this binding
   |
   = note: expected raw pointer `*const Service`
              found raw pointer `*mut u8`
